addons/l10n_ch/models/account_invoice.py

```diff
@@ -189,3 +189,17 @@ def message_post(self, **kwargs):
         if self.env.context.get('l10n_ch_mark_isr_as_sent'):
             self.filtered(lambda inv: not inv.l10n_ch_isr_sent).write({'l10n_ch_isr_sent': True})
         return super(AccountMove, self.with_context(mail_post_autofollow=True)).message_post(**kwargs)
+
+    def _get_invoice_reference_ch_invoice(self):
+        """ This sets ISR reference number which is generated based on customer's `Bank Account` and set it as
+        `Payment Reference` of the invoice when invoice's journal is using Switzerland's communication standard
+        """
+        self.ensure_one()
+        return self.l10n_ch_isr_number_spaced
+
+    def _get_invoice_reference_ch_partner(self):
+        """ This sets ISR reference number which is generated based on customer's `Bank Account` and set it as
+        `Payment Reference` of the invoice when invoice's journal is using Switzerland's communication standard
+        """
```

Comment:
I guess this is WIP.

You probably don't want to have the exact same doc string.

BTW, why do we need that many way to get the same info?

Reply:
Ok got it, the reference is used then in account/models/account_move.py _get_invoice_computed_reference depending on the journal config.

Which will copy the structured ref into `invoice_payment_ref`.